haskell-src/Concordium/Utils.hs

```diff
@@ -23,11 +23,11 @@ class (Ixed m) => At' m where
   at' :: Index m -> Lens' m (Maybe (IxValue m))
 
 instance (Hashable k, Eq k) => At' (H.HashMap k v) where
-  at' = flip H.alterF
+  at' k f = H.alterF f k
```

Comment:
```suggestion
  at' = \k f -> H.alterF f k
```
While the eta-expansion is necessary, I think expressing it as a lambda may give the optimiser more chance of inlining. Lenses in particular are often composed rather than directly applied.

Reply:
Thanks for the suggestion Thomas. On the other I will not use it this time as the linter complains about it.

Reply:
And here is the error message with the original format:

```console
/Users/cemeliguzel/scratch/concordium-client/deps/concordium-base/haskell-src/Concordium/Utils.hs:26:9: error:
    • Couldn't match type: k0
                           -> H.HashMap k0 v0 -> f0 (H.HashMap k0 v0)
                     with: forall (f :: * -> *).
                           Functor f =>
                           (Maybe (IxValue (H.HashMap k v))
                            -> f (Maybe (IxValue (H.HashMap k v))))
                           -> H.HashMap k v -> f (H.HashMap k v)
      Expected: Index (H.HashMap k v)
                -> Lens' (H.HashMap k v) (Maybe (IxValue (H.HashMap k v)))
        Actual: (Maybe v0 -> f0 (Maybe v0))
                -> k0 -> H.HashMap k0 v0 -> f0 (H.HashMap k0 v0)
    • In the expression: H.alterF
      In an equation for ‘at'’: at' = H.alterF
      In the instance declaration for ‘At' (H.HashMap k v)’
    • Relevant bindings include
        at' :: Index (H.HashMap k v)
               -> Lens' (H.HashMap k v) (Maybe (IxValue (H.HashMap k v)))
          (bound at haskell-src/Concordium/Utils.hs:26:3)
   |
26 |   at' = H.alterF
   |         ^^^^^^^^
```